File: random-leetcode/two-sum/java.js
var twoSum = function(nums, target) {
    for(let i = 0; i < nums.length; i++) {
      let currNum = nums[i]
      for(let j = i+1; j < nums.length; j++) {
        let nextNum = nums[j]
        let numSum = currNum + nextNum
        if(target === numSum) {
          return [i, j]
        }
      }
    }
};

File: random-leetcode/two-sum/java.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> pairs = new HashMap<>();

        for(int i = 0; i < nums.length; i++){
            int num = nums[i];
            int diff = target - num;
            if(pairs.containsKey(diff)){
                return new int[] {i, pairs.get(diff)};
            } else {
                pairs.put(num, i);
            }
        }
        return new int[] {};
    }
}
